lib/router/router.js

```diff
@@ -166,7 +172,12 @@ export default class Router {
 
       data = await new Promise((resolve, reject) => {
         this.componentLoadCancel = cancel = () => {
-          if (xhr.abort) xhr.abort()
+          if (xhr.abort) {
+            xhr.abort()
+            const error = new Error('Fetching componenet cancelled')
+            error.cancelled = true
+            reject(error)
```

Comment:
Yep. I didn't see that. If we go that route, we cannot catch this abort errors. I did this because @rauchg mentioned about captured abort events. And I liked that.

If we go on this method, we may need to remove the onabort error callback.